src/fsharp/CompileOps.fs

```diff
@@ -2338,7 +2338,23 @@ let GetAutoOpenAttributes ilg ilModule =
 
 let GetInternalsVisibleToAttributes ilg ilModule = 
     ilModule |> GetCustomAttributesOfIlModule |> List.choose (TryFindInternalsVisibleToAttr ilg)
-    
+
+let CheckOneAssemblyForTypeProviderAttribute (path : string) =
+#if EXTENSIONTYPING
+      let ilGlobals =
+        // "mscorlib" primary assembly can be assumed for the purposes of simply reading off assembly-level attributes
+        mkILGlobals (IL.mkMscorlibBasedTraits ILScopeRef.Local) (Some("mscorlib")) true
+      let opts = { ILBinaryReader.mkDefault ilGlobals with                       
+                      ILBinaryReader.optimizeForMemory=true;
+                      ILBinaryReader.pdbPath = None; } 
+      let reader = ILBinaryReader.OpenILModuleReaderAfterReadingAllBytes path opts
+      GetCustomAttributesOfIlModule reader.ILModuleDef |> List.exists isTypeProviderAssemblyAttr
+#else
+      // prevent unused value warning
+      ignore path
+      false
```

Comment:
is not better to ifdef the CheckOneAssemblyForTypeProviderAttribute function instead of return false?
like this the function return a dummy value, but the caller doesn't know. 
better throw a NotSupportedException if the public surface must remain the same